cogs/utils/parser.py

```diff
@@ -0,0 +1,152 @@
+
+        if len(parsed) == 0:
+            return False
+        try:
+            user = bot.get_user(int(parsed))
+            if user is None:
+                user = await bot.fetch_user(int(parsed))
+        except discord.NotFound:
+            user = None
+
+        return user if user else False
+    else:
+        return False
+
+
+async def channel_parser(channel: Union[discord.TextChannel, discord.VoiceChannel] = None,
+                         ctx=None):
+    """Parses message into a channel"""
+    if isinstance(channel, discord.TextChannel) or isinstance(channel, discord.VoiceChannel):
+        return channel
+    elif not channel and ctx:
+        parsed_channel = ctx.message.author
+        return parsed_channel
+    else:
+        return False
+
+
+async def role_parser(role: Optional[discord.Role]):
+    if isinstance(role, discord.Role):
+        return role
+    else:
+        return False
+
+
+async def human_readable_number(number):
+    number = float(number)
+    pow_ten = 0
+
+    while abs(number) >= 1000:
+        pow_ten += 1
+        number /= 1000
+
+    suffix = ['', 'k', 'm', 'b', 't', "q"][pow_ten]
+
+    if (number*(1000 * pow_ten)) > 1000:
+        return f"{round(number,2)}{suffix}"
+    else:
+        return f"{round(number)}{suffix}"
+
+
+async def timestamp_parser(ctx, message, check=0):
+    converter = nlp_dates.HumanTimeParser(check)
+    return await converter.convert(ctx, message)
+
+
+async def no_ctx_tm_parser(message, check=0):
+    converter = nlp_dates.NoCtxHumanTimeParser(check)
+    return await converter.convert(message)
+
+
+async def remove_special(message, regex=None):
+    if regex:
+        message = re.sub(regex, "", message)
+    return "".join(filter(lambda x: x in string.printable, message))
+
+
+async def back_front_parser(message, length_of_search: int = None, is_list=False):
+    if is_list is False:
+        return [message[:length_of_search], message[-length_of_search:]]
+    elif is_list is True:
+        return [message[0], message[-1]]
+
+        #
+        # when_value = int("".join(re.findall(r"\d+", "".join(timestamp))))
+        # times = "".join(re.findall("[A-Za-z]", "".join(timestamp)))
+        # when_term = conversion_table[times]
+        #
+        # if when_term == "years":
+        #     when_term = "days"
+        #     when_value *= 365
+        #
+        # if ret:
+        #     return (
+        #         datetime.datetime.utcnow() + datetime.timedelta(**{when_term: when_value}),
+        #         " ".join(message).replace(" ".join(timestamp), "")
+        #     )
+        # else:
+        #     return datetime.datetime.utcnow() + datetime.timedelta(**{when_term: when_value})
+
+
+async def human_readable_datetime_parser(ctx, time, dt=True, show_all=False, varient=0):
+    """Converts a datetime object to a human readable version
+    Varient 0 example: 3 hours 50 min
+    Varient 1 example: 3:50:02
+    """
```

Comment:
```suggestion
    """
    if varient not in [0, 1]:
        raise ValueError(f"varient must be 0 or 1, not {varient}")
```
Or whatever error message you want